internal/extender/resource.go

```diff
@@ -519,3 +544,35 @@ func isPodTerminated(pod *v1.Pod) bool {
 	}
 	return allTerminated
 }
+
+// CreateLabelLessThanFunction returns a lessThan function that can be used to order
+// nodes based on the values of a label
+func CreateLabelLessThanFunction(labelPriorityOrder *config.LabelPriorityOrder) func(*resources.NodeSchedulingMetadata, *resources.NodeSchedulingMetadata) bool {
+	if labelPriorityOrder == nil {
+		return nil
+	}
+	valueRanks := make(map[string]int, len(labelPriorityOrder.DescendingPriorityValues))
+	for i, value := range labelPriorityOrder.DescendingPriorityValues {
+		valueRanks[value] = i
+	}
+	return func(metadata1 *resources.NodeSchedulingMetadata, metadata2 *resources.NodeSchedulingMetadata) bool {
+		rank2, ok := extractRank(metadata2.AllLabels, labelPriorityOrder.Name, valueRanks)
+		if !ok {
+			return true
+		}
+		rank1, ok := extractRank(metadata1.AllLabels, labelPriorityOrder.Name, valueRanks)
+		if !ok {
+			return false
+		}
+		return rank1 < rank2
```

Comment:
Should the checks on `rank1` and `rank2` be reversed? Currently:
If both don't have the label, we return `true`.
If both have an equal label rank, we return `false`.
We should pick one and be consistent.